### PR TITLE
differentiate between spatial reference and specification

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts [hunk 1]
```diff
@@ -151,7 +153,7 @@ export class MapContainerComponent implements AfterViewInit, OnChanges, OnDestro
             this.initUserSelect();
 
             this.subscriptions.push(
-                combineLatest([this.mapLayersRaw.changes as Observable<MapLayer>, this.projection$])
+                combineLatest([concat(of({}), this.mapLayersRaw.changes) as Observable<MapLayer>, this.projection$])
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/map/map-container/map-container.component.ts:156]: maybe use startwith?

**Review by jdroenner — CHANGES_REQUESTED** [review]
i think it is fine. i have just some minor changes and questions

File: projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        this.axisLabels = dict.axisLabels;
+    }
+
+    static fromDict(dict: SpatialReferenceSpecificationDict): SpatialReferenceSpecification {
+        return new SpatialReferenceSpecification(dict);
+    }
+}
+
+export class SpatialReference {
+    readonly srsString: SrsString;
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts:29]: we get the spatialReference as a string "authority:code" from the backend. why should I parse it if we never user the one without the other?

File: projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    static fromSrsString(srsString: SrsString): SpatialReference {
+        return new SpatialReference(srsString);
+    }
+}
+
+export class NamedSpatialReference {
+    readonly name: string;
+    readonly spatialReference: SpatialReference;
+
+    constructor(name: string, srsString: SrsString) {
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts:44]: it's less typing to define them below

File: projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        this.spatialReference = SpatialReference.fromSrsString(srsString);
+    }
+}
+
+export const WGS_84 = new NamedSpatialReference('WGS84', 'EPSG:4326');
+export const WEB_MERCATOR = new NamedSpatialReference('WGS84 Webmercator', 'EPSG:3857');
+
+export const UTM32N = new NamedSpatialReference('WGS 84 / UTM 32 N', 'EPSG:32632');
+
+export const WELL_KNOWN_SPATAL_REFERENCES = [WGS_84, WEB_MERCATOR, UTM32N];
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/spatial-references/spatial-reference.model.ts]: c03a493af4ec2341a6f5f408185d282aabc9a762

File: projects/wave-core-new/src/lib/spatial-references/spatial-reference.service.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ * Service for managing spatial references and projections
+ */
+@Injectable()
+export class SpatialReferenceService {
+    private specs = new Map<string, SpatialReferenceSpecification>();
+
+    constructor(protected backend: BackendService, protected userService: UserService) {
+        this.specs.set(
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/spatial-references/spatial-reference.service.ts:42]: c03a493af4ec2341a6f5f408185d282aabc9a762

File: projects/wave-core-new/src/lib/map/map-container/map-container.component.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/map/map-container/map-container.component.ts]: c03a493af4ec2341a6f5f408185d282aabc9a762

**Pull request merged**